protocol/agents.json

```diff
@@ -0,0 +1,85 @@
+{
+    "agents": [
+        {
+            "identifier": "ably-js",
+            "emitters": [
+                "ably-js"
+            ]
+        },
+        {
+            "identifier": "node",
+            "emitters": [
+                "ably-js"
+            ]
+        },
+        {
+            "identifier": "react-native",
+            "versioned": false,
+            "emitters": [
+                "ably-js"
+            ]
+        },
+        {
+            "identifier": "ably-cocoa",
+            "emitters": [
+                "ably-cocoa"
+            ]
+        },
+        {
+            "identifier": "iOS",
+            "emitters": [
+                "ably-cocoa"
+            ]
+        },
+        {
+            "identifier": "tvOS",
+            "emitters": [
+                "ably-cocoa"
+            ]
+        },
+        {
+            "identifier": "watchOS",
+            "emitters": [
+                "ably-cocoa"
+            ]
+        },
+        {
+            "identifier": "macOS",
+            "emitters": [
+                "ably-cocoa"
+            ]
+        },
+        {
+            "identifier": "ably-java",
+            "emitters": [
+                "ably-java"
+            ]
+        },
+        {
+            "identifier": "android",
+            "emitters": [
+                "ably-java"
+            ]
+        },
+        {
+            "identifier": "ably-ruby",
+            "emitters": [
+                "ably-ruby"
+            ]
+        },
+        {
+            "identifier": "ruby",
+            "emitters": [
+                "ably-ruby"
+            ]
+        },
+        {
+            "identifier": "ably-ruby-rest",
+            "versioned": false,
+            "emitters": [
+                "ably-ruby"
+            ]
+        }
+    ]
```

Comment:
Should we include the current agents being sent in `X-Ably-Lib` (e.g. `ably-go`) so we can maintain some continuity in the stats and fix the current issues were having with `X-Ably-Lib` parsing?

Reply:
I agree in principle, although I can't see an obvious way to do this. We'd presumable need to include the name of the `X-Ably-Lib` identifier which will be different than the equivalent `Ably-Agent` identifier and may optionally include a wrapper library. For example it would need to parse `jsrn-1.2.9` as `ably-js/1.2.9 react-native`.

Reply:
For javascript I imagine something like this could work:
```
        {
            "identifier": "ably-js",
            "emitters": [
                "ably-js"
            ],
            "x-ably-lib": {
                "identifier": "js",
                "variants": ["web", "node", "rn", "ns"]
            }
        },
```
This at least describes to the router exactly what the `X-Ably-Lib` is made up of. Actually using this might be more complicated - for example, you would need to account for the fact that `X-Ably-Lib` headers would simply indicate whether NodeJS is being used while `Ably-Agent` will always give you the NodeJS version if present.

Reply:
Hmm ok yes, I didn't realise the X-Ably-Lib version was so ad-hoc.

Looking a bit closer at the current router behaviour, it's even more broken than I first realised since it [splits on the first hyphen](https://github.com/ably/go-services/blob/main/router/request_handler.go#L553), meaning something like `js-web-1.2.9` is currently being emitted as `agent=js version=web-1.2.9` which definitely wasn't the intention.

So, what I'd suggest is defining an explicit mapping from X-Ably-Lib values to their equivalent Ably-Agent value, and use some convention to indicate "no version specified" for Ably-Agent entries which would have a version if provided directly but is missing from the legacy X-Ably-Lib values (e.g. `js-node-1.2.9` maps to `ably-js/1.2.9 nodejs/0.0.0`).

For example, in `agents.json`, have a section like this:

```json
"x-ably-lib-mappings": {
  "js-web":  "ably-js/${version} browser",
  "js-node": "ably-js/${version} nodejs/0.0.0",
  "js-ns":   "ably-js/${version} nativescript",
  "js-rn":   "ably-js/${version} reactnative",
  "go":      "ably-go/${version}"
}
```

and the router would take any `X-Ably-Lib` value that matches `<key>-x.x.x`, lookup `<key>` in the mappings, replace `${version}` with `x.x.x` in the mapped value, then process as if it were an Ably-Agent value.

This makes the handling of X-Ably-Lib pretty clear from just looking at those mappings in `agents.json`, and keeps the router logic pretty generic.

WDYT?

cc @cruickshankpg

Reply:
Related comment: let's not use `kebab-case` for key names - it's only necessary for literal values.

Reply:
added the "x-ably-lib-mappings" property in b7a883e

Reply:
and changed it to camelCase in b7a883e